test_join_sort.py in CI failed by timeout with new Dask release - 2022.2.0

Error: https://github.com/modin-project/modin/runs/5195622251?check_suite_focus=true

Dask release - https://github.com/dask/dask/releases/tag/2022.02.0

Fastest option here - pin `dask<2022.2.0`, but it also requires an investigation into the cause.

> So new Dask release has caused performance to degrade on these tests to the point where they always timeout?

Some kind of memory problem occurs, which leads to a timeout ([example](https://github.com/modin-project/modin/runs/5206389461?check_suite_focus=true)):
```bash
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

>   raise TimeoutError("No valid workers found")
E   asyncio.exceptions.TimeoutError: No valid workers found

C:\Miniconda\envs\modin\lib\site-packages\distributed\scheduler.py:5846: TimeoutError
---------------------------- Captured stderr call -----------------------------
distributed.worker - WARNING - Unmanaged memory use is high. This may indicate a memory leak or the memory may not be released to the OS; see https://distributed.dask.org/en/latest/worker.html#memtrim for more information. -- Unmanaged memory: 448.04 MiB -- Worker memory limit: 476.84 MiB
```

> Is this something that we need to fix, or does it need to be reported to Dask?

I don't know, it's also possible that some `Dask` settings have changed. Can someone take a look? (I don't have the capacity to investigate the problem for now)

--- a/docs/release_notes/release_notes-0.16.0.rst
+++ b/docs/release_notes/release_notes-0.16.0.rst
@@ -72,13 +72,13 @@ Key Features and Updates
 * Dependencies
   * FEAT-#4598: Add support for pandas 1.4.3 (#4599)
   * FEAT-#4619: Integrate mypy static type checking (#4620)
+  * FEAT-#4202: Allow dask past 2022.2.0 (#4769)
 * New Features
   * FEAT-4463: Add experimental fuzzydata integration for testing against a randomized dataframe workflow (#4556)
   * FEAT-#4419: Extend virtual partitioning API to pandas on Dask (#4420)
   * FEAT-#4147: Add partial compatibility with Python 3.6 and pandas 1.1 (#4301)
   * FEAT-#4569: Add error message when `read_` function defaults to pandas (#4647)
   * FEAT-#4725: Make index and columns lazy in Modin DataFrame (#4726)
-  * FEAT-#4202: Allow dask past 2022.2.0 (#4769)
 
 Contributors
 ------------

--- a/docs/release_notes/release_notes-0.16.0.rst
+++ b/docs/release_notes/release_notes-0.16.0.rst
@@ -78,6 +78,7 @@ Key Features and Updates
   * FEAT-#4147: Add partial compatibility with Python 3.6 and pandas 1.1 (#4301)
   * FEAT-#4569: Add error message when `read_` function defaults to pandas (#4647)
   * FEAT-#4725: Make index and columns lazy in Modin DataFrame (#4726)
+  * FEAT-#4202: Allow dask past 2022-2-0 (#4769)
 
 Contributors
 ------------

--- a/docs/release_notes/release_notes-0.16.0.rst
+++ b/docs/release_notes/release_notes-0.16.0.rst
@@ -78,7 +78,7 @@ Key Features and Updates
   * FEAT-#4147: Add partial compatibility with Python 3.6 and pandas 1.1 (#4301)
   * FEAT-#4569: Add error message when `read_` function defaults to pandas (#4647)
   * FEAT-#4725: Make index and columns lazy in Modin DataFrame (#4726)
-  * FEAT-#4202: Allow dask past 2022-2-0 (#4769)
+  * FEAT-#4202: Allow dask past 2022.2.0 (#4769)
 
 Contributors
 ------------

--- a/environment-dev.yml
+++ b/environment-dev.yml
@@ -5,8 +5,8 @@ dependencies:
   - pandas==1.4.3
   - numpy>=1.18.5
   - pyarrow>=4.0.1
-  - dask[complete]>=2.22.0,<2022.2.0
-  - distributed>=2.22.0,<2022.2.0
+  - dask[complete]>=2.22.0
+  - distributed>=2.22.0
   - fsspec
   - xarray
   - Jinja2

--- a/requirements-dev.txt
+++ b/requirements-dev.txt
@@ -1,8 +1,8 @@
 pandas==1.4.3
 numpy>=1.18.5
 pyarrow>=4.0.1
-dask[complete]>=2.22.0,<2022.2.0
-distributed>=2.22.0,<2022.2.0
+dask[complete]>=2.22.0
+distributed>=2.22.0
 ray[default]>=1.4.0
 redis>=3.5.0,<4.0.0
 psutil

--- a/requirements/requirements-py36.txt
+++ b/requirements/requirements-py36.txt
@@ -1,8 +1,8 @@
 pandas==1.1.5
 numpy>=1.18.5
 pyarrow>=4.0.1
-dask[complete]>=2.22.0,<2022.2.0
-distributed>=2.22.0,<2022.2.0
+dask[complete]>=2.22.0
+distributed>=2.22.0
 ray[default]>=1.4.0
 redis>=3.5.0,<4.0.0
 psutil

--- a/setup.py
+++ b/setup.py
@@ -7,7 +7,7 @@ PANDAS_VERSION = "1.4.3" if sys.version_info >= (3, 8) else "1.1.5"
 with open("README.md", "r", encoding="utf-8") as fh:
     long_description = fh.read()
 
-dask_deps = ["dask>=2.22.0,<2022.2.0", "distributed>=2.22.0,<2022.2.0"]
+dask_deps = ["dask>=2.22.0", "distributed>=2.22.0"]
 if sys.version_info < (3, 8):
     dask_deps.append("pickle5")
 

